### PR TITLE
Unzip RC file use case

### PR DESCRIPTION
**Pull request by mbr4477**
- New UnZipResourceContainer use case to unzip RC into app data directory before importing
- Updated ImportResourceContainer and CreateProject to take advantage of new dao/repo transaction behaviors for faster import/create

### CODE REVIEW

File: src/main/kotlin/org/wycliffeassociates/otter/common/domain/ImportResourceContainer.kt [hunk 1]
```diff
@@ -18,52 +18,43 @@ import java.io.File
...
     fun import(file: File): Completable {
         return when {
             file.isDirectory -> importDirectory(file)
+            file.extension.toLowerCase() == "zip" -> Single
+                    .fromCallable {
+                        UnZipResourceContainer(file, rcDirectory.absoluteFile).unzip()
+                    }
+                    .flatMapCompletable {
+                        importDirectory(it)
+                    }
```
Comment (jsarabia) [inline src/main/kotlin/org/wycliffeassociates/otter/common/domain/ImportResourceContainer.kt:39]: wrap this in an importZip function

File: src/main/kotlin/org/wycliffeassociates/otter/common/domain/ImportResourceContainer.kt [hunk 2]
```diff
@@ -76,26 +67,151 @@ class ImportResourceContainer(
...
+                                            .insert(resourceMetadata)
+                                            .map {
+                                                resourceMetadata.id = it
+                                                return@map resourceMetadata
+                                            }
+                                }
+                                .flatMapCompletable { resourceMetadata ->
+                                    // Create empty lists to hold the testaments
+                                    val oldTestamentBooks = mutableListOf<RelatedCollection>()
+                                    val newTestamentBooks = mutableListOf<RelatedCollection>()
```
Comment (jsarabia) [inline src/main/kotlin/org/wycliffeassociates/otter/common/domain/ImportResourceContainer.kt:96]: don't hard code bible stuff into importing a resource container

**Review by jsarabia — CHANGES_REQUESTED** [review]
I'm okay to pull in unzipping a resource container, but not the changes to importing to the database- this needs to be sql anyway, but I don't want anything to be context specific, it needs to be generic to a resource container.

**Comment by mbr4477** [comment]
Architecture change

**Pull request closed without merging**